presto-main/src/test/java/com/facebook/presto/sql/planner/TestAddNotNullFiltersToJoinNode.java

```diff
@@ -0,0 +1,242 @@
+import static com.facebook.presto.sql.planner.plan.JoinNode.Type.LEFT;
+import static com.facebook.presto.sql.planner.plan.JoinNode.Type.RIGHT;
+import static com.facebook.presto.sql.relational.Expressions.variable;
+import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+
+public class TestAddNotNullFiltersToJoinNode
+        extends BasePlanTest
+{
+    private final Map<String, Type> testVariableTypeMap = ImmutableMap.of("a", BIGINT, "b", BIGINT, "c", BIGINT, "d", BIGINT, "e", BIGINT);
+    private final ExtractInferredNotNullVariablesVisitor visitor;
+    private final TestingRowExpressionTranslator rowExpressionTranslator;
+
+    public TestAddNotNullFiltersToJoinNode()
+    {
+        super(ImmutableMap.of(OPTIMIZE_NULLS_IN_JOINS_V2, Boolean.toString(true),
+                OPTIMIZE_NULLS_IN_JOINS, Boolean.toString(false)));
+
+        FunctionAndTypeManager functionAndTypeManager = createTestFunctionAndTypeManager();
+        Metadata metadata = MetadataManager.createTestMetadataManager();
+        rowExpressionTranslator = new TestingRowExpressionTranslator(metadata);
+        visitor = new ExtractInferredNotNullVariablesVisitor(functionAndTypeManager);
+    }
+
+    @Test
+    public void testNotNullPredicatesAddedForSingleEquiJoinClause()
+    {
+        String query = "select 1 from lineitem l join orders o on l.orderkey = o.orderkey";
+        assertPlan(query,
+                anyTree(
+                        join(INNER,
+                                ImmutableList.of(equiJoinClause("LINE_ORDER_KEY", "ORDERS_ORDER_KEY")),
+                                anyTree(
+                                        filter("LINE_ORDER_KEY IS NOT NULL",
+                                                tableScan("lineitem", ImmutableMap.of("LINE_ORDER_KEY", "orderkey")))),
+                                anyTree(
+                                        filter("ORDERS_ORDER_KEY IS NOT NULL",
+                                                tableScan("orders", ImmutableMap.of("ORDERS_ORDER_KEY", "orderkey")))))));
+    }
+
+    @Test
+    public void testNotNullPredicatesAddedForCrossJoinReducedToInnerJoin()
+    {
+        String query = "select 1 from lineitem l, orders o where l.orderkey = o.orderkey";
+        assertPlan(query,
+                anyTree(
+                        join(INNER,
+                                ImmutableList.of(equiJoinClause("LINE_ORDER_KEY", "ORDERS_ORDER_KEY")),
+                                anyTree(
+                                        filter("LINE_ORDER_KEY IS NOT NULL",
+                                                tableScan("lineitem", ImmutableMap.of("LINE_ORDER_KEY", "orderkey")))),
+                                anyTree(
+                                        filter("ORDERS_ORDER_KEY IS NOT NULL",
+                                                tableScan("orders", ImmutableMap.of("ORDERS_ORDER_KEY", "orderkey")))))));
```

Comment:
Identical test as in testNotNullPredicatesAddedForSingleEquiJoinClause?

Reply:
In this test we use the [older join syntax](https://flylib.com/books/en/4.343.1.38/1/) of having `from a,b where a.col = b.col` which translates initially to a cross join node and is reduced later to an inner join. The older code (#14578) was not able to add not-null filters for such joins